README.md

```diff
@@ -2,6 +2,113 @@
 
 # prpl-server-node
 
-A Node implementation of the [PRPL](https://developers.google.com/web/fundamentals/performance/prpl-pattern/) pattern for serving Progressive Web Apps.
+An HTTP server for Node designed for serving [PRPL](https://developers.google.com/web/fundamentals/performance/prpl-pattern/) apps in production.
 
-**(Note: This project is in early development and is not yet suitable for use.)**
+## Usage
+
+### As a binary
+```sh
+npm install -g prpl-server
+prpl-server
+```
+
+### As a library
+Yarn is recommended, but npm works too.
+
+```sh
+yarn add prpl-server
+```
+
+```js
+prpl = require('prpl-server');
+http = require('http');
+http.createServer(prpl.makeHandler()).listen(8080);
+```
+
+## Differential Serving
+
+Modern browsers offer great features that improve performance, but most applications need to support older browsers too. prpl-server can serve different versions of your application to different browsers by detecting browser capabilities using the user-agent header.
+
+### Builds
+
+prpl-server understands the notion of a *build*, a variant of your application optimized for a particular set of browser capabilities.
+
+Builds are specified in a JSON configuration file. This format is compatible with [`polymer.json`](https://www.polymer-project.org/2.0/docs/tools/polymer-json), so if you are already using polymer-cli for your build pipeline, you can annotate your existing builds with browser capabilities, and copy the configuration to your server root. prpl-server will look for a file called `polymer.json` in the server root, or you can specify it directly with the `--config` flag.
+
+
+In this example we define two builds, one for modern browsers that support ES2015 and HTTP/2 Push, and a fallback build for other browsers:
+
+```
+{
+  "entrypoint: "index.html",
+  "builds": {
+    {
+      "name": "modern",
+      "browserCapabilities": ["push", "es2015"]
+    },
+    {
+      "name": "fallback",
+      "browserCapabilities": []
+    }
+  }
+}
+```
+
+### Capabilities
+
+The `browserCapabilities` field defines the browser features required for that build. prpl-server analyzes the request user-agent header and picks the best build for which all capabilities are met. If multiple builds are compatible, the one with more capabilities is preferred. If there is a tie, the build that comes earlier in the configuration file wins.
+
+You should always include a fallback build with no capability requirements. If you don't, prpl-server will warn at startup, and will return a 400 error on entrypoint requests to browsers for which no build can be served.
+
+## Entrypoint
+
+In the [PRPL pattern](https://developers.google.com/web/fundamentals/performance/prpl-pattern/), the *entrypoint* is a small HTML file that acts as the application bootstrap.
+
+prpl-server will serve the entrypoint from the best compatible build from `/`, and from any path that does not have a file extension and is not an existing file.
```

Comment:
This brings up an interesting situation where routing could be different based on UA